docs/security-privacy-self-review.md

```diff
@@ -0,0 +1,85 @@
+
+## 2.4 How does this specification deal with sensitive information?
+
+As in the previous question, the specification does not limit what data is shared, and the data could include sensitive information already revealed to the web site. However, the user is in control of which, if any, target application will receive the information.
+
+## 2.5 Does this specification introduce new state for an origin that persists across browsing sessions?
+
+No.
+
+## 2.6 What information from the underlying platform, e.g. configuration data, is exposed by this specification to an origin?
+
+The presence of the API may reveal characteristics of the user's device beyond the user agent. For example, the user agent may only expose the `navigator.share()` API on "mobile" devices, revealing the type of device the user agent is executing on.
+
+Revealing content types supported via `files` should be done carefully, as `canShare()` does not require
+[transient activation](https://html.spec.whatwg.org/multipage/interaction.html#transient-activation). For example, if the `canShare()` API response was conditional on what native applications were installed on the device, this could be used to fingerprint the user.
+
+## 2.7 Does this specification allow an origin access to sensors on a user’s device
+
+No.
+
+## 2.8 What data does this specification expose to an origin? Please also document what data is identical to data exposed by other features, in the same or different contexts.
+
+The data exposed to an origin is the presence of the API itself, whether certain content types are supported, and the results of interacting with the API (error or success).
+
+## 2.9 Does this specification enable new script execution/loading mechanisms?
+
+No.
+
+## 2.10 Does this specification allow an origin to access other devices?
+
+No.
+
+## 2.11 Does this specification allow an origin some measure of control over a user agent’s native UI?
+
+The API provides for showing a target selection user interface, for example as a modal dialog. The presence and presentation of the interface remains entirely under the control of the user agent.
+
+## 2.12 What temporary identifiers might this this specification create or expose to the web?
+
+None.
+
+## 2.13 How does this specification distinguish between behavior in first-party and third-party contexts?
+
+First-party and third-party contexts are not distingushed.
+
+The use of the `share()` API is gated on [transient activation](https://html.spec.whatwg.org/multipage/interaction.html#transient-activation) and thus requires user interaction (or the equivalent) within a third-party context to be used.
+
```

Comment:
@mgiuca, I can't recall, but do is the API disabled by default in third-party contexts (i.e., only enabled by feature policy)?

Reply:
I don't think there's any such restriction (either in the spec, or in Chrome's implementation).

Not sure what "requires user interaction (or the equivalent) within a third-party context to be used" is getting at. It always requires user interaction, why call out "within a third-party context" here?

Reply:
It is phrased this way to make it clear that even though third-party iframes can use `share()` - which might be surprising - the user still must have interacted with the frame. So an embedded ad frame is not going to be able to annoy the user, unless the user has interacted with the frame itself.

Reply:
(I admit to being surprised that we don't require feature policy for 3P contexts when I read the spec.)

Reply:
Agree. I think we should treat this as a bug.

Reply:
resolved based on us having filed https://github.com/w3c/web-share/issues/151